server/container_remove.go

```diff
@@ -34,6 +34,13 @@ func (s *Server) RemoveContainer(ctx context.Context, req *pb.RemoveContainerReq
 		return nil, fmt.Errorf("failed to delete container %s: %v", c.ID(), err)
 	}
 
+	s.releaseContainerName(c.Name())
```

Comment:
I think we can keep the removing in-memory state last. That way we can support idempotent operations and clear them from in-memory state only once the container is removed and storage is cleaned up. WDYT?